lib/ansible/modules/windows/win_firewall_profile.ps1

```diff
@@ -0,0 +1,60 @@
+#!powershell
+
+# Copyright : (c) 2019, Simon Lecoq <simon.lecoq@live.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+$firewall_profiles = @("Domain", "Private", "Public")
+$firewall_actions = @("NotConfigured", "", "Allow", "", "Block")
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$profiles = Get-AnsibleParam -obj $params -name "profiles" -type "list" -default $firewall_profiles
+$inbound = Get-AnsibleParam -obj $params -name "inbound" -type "str" -validateset 'NotConfigured','Allow','Block'
+$outbound = Get-AnsibleParam -obj $params -name "outbound" -type "str" -validateset 'NotConfigured','Allow','Block'
+
+$result = @{
+    changed = $false
+    profiles = $profiles
+    inbound = $inbound
+    outbound = $outbound
+}
+
+Try {
+
+    ForEach ($profile in $firewall_profiles) {
+
+        $currentinbound = (Get-NetFirewallProfile -Name $profile).DefaultInBoundAction
+        $currentoutbound = (Get-NetFirewallProfile -Name $profile).DefaultOutBoundAction
+        $result.$profile = @{
+            considered = ($profiles -contains $profile)
+            inbound = $firewall_actions[$currentinbound]
+            outbound = $firewall_actions[$currentoutbound]
+        }
+
+        if ($profiles -notcontains $profile) {
```

Comment:
IMHO continue makes things look weird in the code, think about just using the above check you make for considered with the two later ifs in it